Makefile

```diff
@@ -968,14 +986,14 @@ $(TARGET_ELF):  $(TARGET_OBJS)
 # Compile
 $(OBJECT_DIR)/$(TARGET)/%.o: %.c
 	$(V1) mkdir -p $(dir $@)
-	$(V1) $(if $(findstring $(subst ./src/main/,,$<), $(SPEED_OPTIMISED_SRC)), \
+	$(V1) $(if $(findstring $(notdir $<),$(SPEED_OPTIMISED_SRC)), \
```

Comment:
I don't think it makes sense to replace something that  is correct functionally but has cosmetic issues with something that we know is functionally wrong. I've got an idea on how to get something that is functionally and semantically correct, will submit a pr when I'm in the office.